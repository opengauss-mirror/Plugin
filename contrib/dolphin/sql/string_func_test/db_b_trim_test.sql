set dolphin.b_compatibility_mode to off;
drop database if exists db_b_trim_test;
create database db_b_trim_test dbcompatibility 'A';
\c db_b_trim_test

SELECT TRIM('  bar   ');
SELECT TRIM(LEADING 'x' FROM 'xxxbarxxx');
SELECT TRIM(BOTH 'x' FROM 'xxxbarxxx');
SELECT TRIM(TRAILING 'x' FROM 'xxxbarxxx');
SELECT TRIM(LEADING 'xyz' FROM 'xyzxbarxxyz');
SELECT TRIM(BOTH 'xyz' FROM 'xyzxbarxxyz');
SELECT TRIM(TRAILING 'xyz' FROM 'xyzxbarxxyz');

\c contrib_regression
set dolphin.b_compatibility_mode to off;
drop database db_b_trim_test;
set dolphin.b_compatibility_mode to on;

create schema db_b_trim_test;
set current_schema to 'db_b_trim_test';

SELECT TRIM('  bar   ');
SELECT TRIM(LEADING 'x' FROM 'xxxbarxxx');
SELECT TRIM(BOTH 'x' FROM 'xxxbarxxx');
SELECT TRIM(TRAILING 'x' FROM 'xxxbarxxx');
SELECT TRIM(LEADING 'xyz' FROM 'xyzxbarxxyz');
SELECT TRIM(BOTH 'xyz' FROM 'xyzxbarxxyz');
SELECT TRIM(TRAILING 'xyz' FROM 'xyzxbarxxyz');

set bytea_output to escape;

SELECT TRIM(LEADING 'x'::bytea FROM 'xxxbarxxx'::bytea);
SELECT TRIM(BOTH 'x'::bytea FROM 'xxxbarxxx'::bytea);
SELECT TRIM(TRAILING 'x'::bytea FROM 'xxxbarxxx'::bytea);
SELECT TRIM(LEADING 'xyz'::bytea FROM 'xyzxbarxxyz'::bytea);
SELECT TRIM(BOTH 'xyz'::bytea FROM 'xyzxbarxxyz'::bytea);
SELECT TRIM(TRAILING 'xyz'::bytea FROM 'xyzxbarxxyz'::bytea);

SELECT TRIM('   X  '::bytea);
SELECT TRIM(LEADING '   X  '::bytea);
SELECT TRIM(TRAILING '   X  '::bytea);

drop schema db_b_trim_test cascade;
reset current_schema;