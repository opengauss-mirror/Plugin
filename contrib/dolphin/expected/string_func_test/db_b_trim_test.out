SELECT TRIM('  bar   ');
 btrim 
-------
 bar
(1 row)

SELECT TRIM(LEADING 'x' FROM 'xxxbarxxx');
 ltrim  
--------
 barxxx
(1 row)

SELECT TRIM(BOTH 'x' FROM 'xxxbarxxx');
 btrim 
-------
 bar
(1 row)

SELECT TRIM(TRAILING 'x' FROM 'xxxbarxxx');
 rtrim  
--------
 xxxbar
(1 row)

SELECT TRIM(LEADING 'xyz' FROM 'xyzxbarxxyz');
  ltrim  
---------
 barxxyz
(1 row)

SELECT TRIM(BOTH 'xyz' FROM 'xyzxbarxxyz');
 btrim 
-------
 bar
(1 row)

SELECT TRIM(TRAILING 'xyz' FROM 'xyzxbarxxyz');
  rtrim  
---------
 xyzxbar
(1 row)

drop database if exists test;
NOTICE:  database "test" does not exist, skipping
create database test dbcompatibility 'B';
\c test
SELECT TRIM('  bar   ');
 trim 
------
 bar
(1 row)

SELECT TRIM(LEADING 'x' FROM 'xxxbarxxx');
  trim  
--------
 barxxx
(1 row)

SELECT TRIM(BOTH 'x' FROM 'xxxbarxxx');
 trim 
------
 bar
(1 row)

SELECT TRIM(TRAILING 'x' FROM 'xxxbarxxx');
  trim  
--------
 xxxbar
(1 row)

SELECT TRIM(LEADING 'xyz' FROM 'xyzxbarxxyz');
   trim   
----------
 xbarxxyz
(1 row)

SELECT TRIM(BOTH 'xyz' FROM 'xyzxbarxxyz');
 trim  
-------
 xbarx
(1 row)

SELECT TRIM(TRAILING 'xyz' FROM 'xyzxbarxxyz');
   trim   
----------
 xyzxbarx
(1 row)

\c postgres
drop database test;
